Sources/SwiftKafka/New/ConsumerConfig.swift

```diff
@@ -0,0 +1,456 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-gsoc open source project
+//
+// Copyright (c) 2022 Apple Inc. and the swift-kafka-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+public struct ConsumerConfig: Hashable, Equatable, StringDictionaryRepresentable {
+    var dictionary: [String: String] = [:]
+
+    // MARK: - Consumer-specific Config Properties
+
+    /// Client group id string. All clients sharing the same group.id belong to the same group.
+    public var groupID: String {
+        get { self.dictionary["group.id"] ?? "" }
+        set { self.dictionary["group.id"] = String(newValue) }
+    }
+
+    /// Client group session and failure detection timeout. The consumer sends periodic heartbeats (heartbeat.interval.ms) to indicate its liveness to the broker. If no hearts are received by the broker for a group member within the session timeout, the broker will remove the consumer from the group and trigger a rebalance. The allowed range is configured with the broker configuration properties group.min.session.timeout.ms and group.max.session.timeout.ms. Also see max.poll.interval.ms.
+    public var sessionTimeoutMs: UInt {
+        get { self.dictionary.getUInt("session.timeout.ms") ?? 45000 }
```

Comment:
Are all the default values here returned the ones that librdkafka uses?

Reply:
Yes! I used [librdkafka/CONFIGURATION.md](https://github.com/confluentinc/librdkafka/blob/master/CONFIGURATION.md) for reference.